### PR TITLE
Add docs for using Mariner-based images

### LINKED ISSUE #2514: Document support for Mariner-based images
**What content needs to be created or modified?**

Starting with Dapr 1.8, we are publishing images based on Mariner: https://github.com/dapr/dapr/pull/4707

We need to document what they are and how to use them

**Where should the new material be placed?**

@msfussell where do you recommend these docs to go?

**The associated pull request from dapr/dapr, dapr/components-contrib, or other Dapr code repos**

https://github.com/dapr/dapr/pull/4707

### PR DESCRIPTION
**Pull request by ItalyPaleAle**
Fixes #2514

### CODE REVIEW

File: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-deploy.md [hunk 1]
```diff
@@ -171,6 +171,25 @@ dapr-sidecar-injector-8555576b6f-29cqm   1/1       Running   0          40s
 dapr-sentry-9435776c7f-8f7yd             1/1       Running   0          40s
 ```
 
+## Using Mariner-based images
+
+When deploying Dapr, on Kubernetes but also in Docker self-hosted, the default container images that are pulled are based on [*distroless*](https://github.com/GoogleContainerTools/distroless).
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/hosting/kubernetes/kubernetes-deploy.md]:
```suggestion
When deploying Dapr, whether on Kubernetes or in Docker self-hosted, the default container images that are pulled are based on [*distroless*](https://github.com/GoogleContainerTools/distroless).
```

File: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-deploy.md [hunk 2]
```diff
@@ -171,6 +171,25 @@ dapr-sidecar-injector-8555576b6f-29cqm   1/1       Running   0          40s
...
+
+As an alternative option, you can use Dapr container images based on Mariner 2 (minimal distroless). [Mariner](https://github.com/microsoft/CBL-Mariner/), officially known as CBL-Mariner, is a free and open source Linux distribution and container base image that is maintained by Microsoft. For some Dapr users, leveraging container images based on Mariner can help meeting compliance requirements.
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/hosting/kubernetes/kubernetes-deploy.md]:
```suggestion
Alternatively, you can use Dapr container images based on Mariner 2 (minimal distroless). [Mariner](https://github.com/microsoft/CBL-Mariner/), officially known as CBL-Mariner, is a free and open-source Linux distribution and container base image maintained by Microsoft. For some Dapr users, leveraging container images based on Mariner can help them meet compliance requirements.
```

File: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-deploy.md [hunk 3]
```diff
@@ -171,6 +171,25 @@ dapr-sidecar-injector-8555576b6f-29cqm   1/1       Running   0          40s
...
+
+To use Mariner-based images for Dapr, you need to add `-mariner` to your Docker tags. For example, while `ghcr.io/dapr/dapr:latest` is the Docker image based on *distroless*, `ghcr.io/dapr/dapr:latest-mariner` is based on Mariner. Tags pinned to a specific version are available too, such as `{{% dapr-latest-version short="true" %}}-mariner`.
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/hosting/kubernetes/kubernetes-deploy.md]:
```suggestion
To use Mariner-based images for Dapr, you need to add `-mariner` to your Docker tags. For example, while `ghcr.io/dapr/dapr:latest` is the Docker image based on *distroless*, `ghcr.io/dapr/dapr:latest-mariner` is based on Mariner. Tags pinned to a specific version are also available, such as `{{% dapr-latest-version short="true" %}}-mariner`.
```

**Review by hhunter-ms — CHANGES_REQUESTED** [review]

File: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-deploy.md (hunk 2 above)
Comment (ItalyPaleAle) [reply daprdocs/content/en/operations/hosting/kubernetes/kubernetes-deploy.md]: How about "help you meet"?

**Comment by ItalyPaleAle** [comment]
Thanks for the review @hhunter-ms :)

File: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-deploy.md (hunk 2 above)
Comment (hhunter-ms) [reply daprdocs/content/en/operations/hosting/kubernetes/kubernetes-deploy.md]: That works for me too!

**Review by hhunter-ms — APPROVED** [review]

**Pull request merged**